EIPS/eip-7892.md

```diff
@@ -59,37 +71,107 @@ To facilitate these changes on the execution layer, the `blobSchedule` object sp
     MAX_BLOBS_PER_BLOCK: 24
-  - EPOCH: 355368
+  - EPOCH: 420000     ## A future anonymous BPO fork
     MAX_BLOBS_PER_BLOCK: 56
+  - EPOCH: 440000     ## GLOAS_FORK_EPOCH; a future named fork introducing blob parameter changes
+    MAX_BLOBS_PER_BLOCK: 72
 ```
 
 The parameters and schedules above are purely illustrative. Actual values and schedules are beyond the scope of this specification.
 
-### Requirements
+**Requirements:**
+
+- Execution and consensus clients **MUST** share consistent BPO fork schedules.
+- The slot number in the EL's `blobSchedule` **MUST** align with the start of the epoch specified in the consensus layer configuration.
+- The `max` field in the EL's `blobSchedule` **MUST** equal the `MAX_BLOBS_PER_BLOCK` value in the consensus layer configuration.
+
+### Modified `compute_fork_digest`
+
+The `compute_fork_digest` helper is updated to account for BPO forks:
+
+```python
+# These basic types already exist. Copied here for reference.
+Epoch = int
+Version = bytes
+Root = bytes
+ForkDigest = bytes
+
+class BlobScheduleEntry(NamedTuple):
+   epoch: Epoch
+   max_blobs_per_block: int       # uint32
+
+def compute_fork_digest(
+  current_version: Version,       # Unchanged. Refers to the baseline hardfork atop which the blob schedule is applied.
+  genesis_validators_root: Root,  # Unchanged.
+  current_epoch: Epoch,           # New.
+  blob_schedule: Sequence[BlobScheduleEntry]  # New.
+) -> ForkDigest:
+    """
+    Return the 4-byte fork digest for the ``current_version`` and ``genesis_validators_root``,
+    bitmasking blob parameters after ``ELECTRA_FORK_VERSION``.
+
+    This is a digest primarily used for domain separation on the p2p layer.
+    4-bytes suffices for practical separation of forks/chains.
+    """
+    base_digest = compute_fork_data_root(current_version, genesis_validators_root)[:4]
+
+    # Find the blob parameters applicable to this epoch.
+    sorted_schedule = sorted(blob_schedule, key=lambda e: e.epoch, reverse=True)
+    blob_params = next(
+      (entry for entry in sorted_schedule if current_epoch >= entry.epoch),
+      None
+    )
+
+    # This check enables us to roll out the BPO mechanism without a concurrent parameter change.
+    if blob_params is None:
+      return ForkDigest(base_digest)
+
+    # Bitmask blob parameters into the digest.
+    mask = blob_params.max_blobs_per_block.to_bytes(4, 'big')
+    masked_digest = bytes(a ^ b for a, b in zip(base_digest, mask))
+    return ForkDigest(masked_digest)
+```
+
+### P2P Networking
+
+#### ENR
+
+In the consensus layer, ENRs are extended with an additional entry that communicates the digest of the next scheduled fork, regardless of whether it is a regular or BPO fork. This approach is preferred over encoding BPO-specific parameters because it is agnostic to specific use cases and offers greater long-term flexibility.
+
+| Key     | Value                   |
+| :-----  | :---------------------- |
+| `nfd`   | SSZ Bytes4 `ForkDigest` |
```

Comment:
It might be worth explicitly calling out that `nfd` stands for "next fork digest" somewhere in this section.

Reply:
Oops, this was defined but got edited away. Will re-add.